backend/coreapp/compiler_wrapper.py

```diff
@@ -239,15 +245,18 @@ def get_objdump_command(platform: str) -> Optional[str]:
         return _platforms[platform].objdump_cmd
     return None
 
-def _check_compilation_cache(*args: str) -> Tuple[Optional[Compilation], str]:
-    hash = util.gen_hash(args)
-    return Compilation.objects.filter(hash=hash).first(), hash
-
 def _check_assembly_cache(*args: str) -> Tuple[Optional[Assembly], str]:
     hash = util.gen_hash(args)
     return Assembly.objects.filter(hash=hash).first(), hash
 
 
+class CompilationResult:
+    def __init__(self, compilation: Optional[Compilation], elf_object: bytes, errors: str):
+        self.compilation = compilation
+        self.elf_object = elf_object
+        self.errors = errors
```

Comment:
This would be a great use of `@dataclass`, like `Platform` in `compiler_wrapper.py`